pyramid/scripts/pserve.py

```diff
@@ -887,11 +889,28 @@ def check_reload(self):
                 continue
             if filename.endswith('.pyc') and os.path.exists(filename[:-1]):
                 mtime = max(os.stat(filename[:-1]).st_mtime, mtime)
+                pyc = True
+            else:
+                pyc = False
             if filename not in self.module_mtimes:
                 self.module_mtimes[filename] = mtime
             elif self.module_mtimes[filename] < mtime:
-                print("%s changed; reloading..." % filename)
-                return False
+                if pyc:
+                    filename = filename[:-1]
+                # check if a file has syntax errors. If so, track it until it's fixed
+                for filename_promise in self.syntax_error_files + [filename]:
+                    try:
+                        py_compile.compile(filename_promise, doraise=True)
+                    except py_compile.PyCompileError:
+                        print("%s has a SyntaxError; NOT reloading." % filename_promise)
```

Comment:
I think we should be printing SyntaxErrors on each reload. Not-so-small applications will quickly configure logging for their app.

Consider a following common use case:

- Developer changes file `foo.py` and makes a SyntaxError
- pserve prints the error message and doesn't reload
- Developer goes to the browser and reloads to see the new code
- Browsers makes requests to the server, prints out lots of logging about them
- Developer is confused and checks the console, but due to a bunch of logging, syntax error is hidden in history

TL;DR: printing out the error on each reload is annoying, but so it making a SyntaxError.